jira-clone/Toast.jsx

```diff
@@ -0,0 +1,51 @@
+// Source: https://github.com/oldboyxx/jira_clone/blob/26a9e77b1789fef9cb43edb5d6018cf1663cf035/client/src/App/Toast/index.jsx#L1
+import React, { useState, useEffect } from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import pubsub from 'sweet-pubsub';
+import { uniqueId } from 'lodash';
+
+import { Container, StyledToast, CloseIcon, Title, Message } from './Styles';
+
+const Toast = () => {
+  const [toasts, setToasts] = useState([]);
+
+  useEffect(() => {
+    const addToast = ({ type = 'success', title, message, duration = 5 }) => {
```

Comment:
> [!WARNING]
> <ins>Don't redeclare something unless necessary</ins>: This can cause inefficiencies because if this effect is run multiple times then this function will be declared for every run. I'm marking this comment as a warning (yellow) and not an error (red) because by giving `useEffect` an empty list of dependencies, this effect will only run once i.e on-component-did-mount. However, that isn't explicit and (probably) takes some time to realize. I personally would declare this function outside the effect and probably memoize it using `useCallback`.